### PR TITLE
Stop passing rule to interception listener

### LINKED ISSUE #1494: Stop passing IFakeObjectCallRule to OnAfterCallIntercepted in 5.0.0
In https://github.com/FakeItEasy/FakeItEasy/pull/1491#discussion_r233072656, @blairconrad said:

> As I hint at, this hack wouldn't be necessary if `IFakeObjectCallRule` didn't expose `Apply`. But it needs to in order for people to create their own rules.
> 
> But I don't think that `IInterceptionListener.OnAfterCallIntercepted` needs to be passed an object that has an `Apply` method; there's no need for `OnAfterCallIntercepted` to call `Apply`.
> I don't even know why it would need to call `IsApplicable` or `NumberOfTimesToCall`. Shall we explore changing this in a future major version? We could supply `OnAfterCallIntercepted` with a different interface that's a restricted version of `IFakeObjectCallRule`, or even remove the `IFakeObjectCallRule` parameter.

@thomaslevesque replied:

> Seems like a good idea. I don't see any value in having the call rule available in `OnAfterCallIntercepted`, except for information purposes. Maybe we could just pass a string describing the rule, or just the type of the rule.

### PR DESCRIPTION
**Pull request by thomaslevesque**
Fixes #1494

### CODE REVIEW

**Comment by thomaslevesque** [comment]
Oops, looks like I messed up my branches...

**Comment by thomaslevesque** [comment]
Fixed

The first commit passes the type of the rule. The second commit removes the parameter completely.

**Comment by blairconrad** [comment]
> Oops, looks like I messed up my branches...

Ha. This morning I accidentally pushed to origin/develop, then reset local to upstream/develop, pushed, realized I'd lost all my work, and restored from reflog. 
On the brighter side, a colleague just learning git had had some questions about whether commits disappear when they're no longer reachable, so I was able to copy my fiasco to use as a teaching moment on Monday.

File: tests/FakeItEasy.Tests/Core/FakeManagerTests.cs [hunk 1]
```diff
@@ -394,7 +394,7 @@ public void Should_invoke_listener_after_call_has_been_intercepted()
             ProcessFakeObjectCall(manager, interceptedCall);
 
             // Assert
-            A.CallTo(() => listener.OnAfterCallIntercepted(completedCall, selectedRule)).MustHaveHappened();
+            A.CallTo(() => listener.OnAfterCallIntercepted(completedCall)).MustHaveHappened();
```
Comment (blairconrad) [inline tests/FakeItEasy.Tests/Core/FakeManagerTests.cs]: We could remove `selectedRule` altogether. We don't _have to_, but we could.

Comment (blairconrad) [reply tests/FakeItEasy.Tests/Core/FakeManagerTests.cs]: I'm not trying to be pushy, but usually you'd comply or comment after I say something like this, and you did neither. I suspect you missed my late-breaking comment.

Comment (thomaslevesque) [reply tests/FakeItEasy.Tests/Core/FakeManagerTests.cs]: I'm not sure I understand... I did remove `selectedRule`!

Comment (thomaslevesque) [reply tests/FakeItEasy.Tests/Core/FakeManagerTests.cs]: Ah, you mean the variable is still there! Sorry, will fix.

**Comment by blairconrad** [comment]
Whoops. I didn't notice your latest commit. Great. Thanks!

**Comment by thomaslevesque** [comment]
Squashed

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque.

**Pull request merged**